qubesfiletrust/qvm_file_trust.py

```diff
@@ -0,0 +1,546 @@
+    path = os.path.normpath(path)
+
+    global ALL_PATHS_ARE_UNTRUSTED
+    global UNTRUSTED_PATH_FOUND
+
+    # Check if path is in the untrusted paths list
+    if is_untrusted_path(path):
+        # Print out which paths are untrusted if we're checking multiple paths
+        if multiple_paths:
+            qprint('Untrusted: {}'.format(path), False)
+            UNTRUSTED_PATH_FOUND = True
+        else:
+            qprint('Folder is untrusted', False)
+            sys.exit(1)
+    else:
+        # Don't return until we've checked all paths
+        if multiple_paths:
+            ALL_PATHS_ARE_UNTRUSTED = False
+        else:
+            qprint('Folder is trusted', False)
+            sys.exit(0)
+
+def change_file(path, trusted):
+    """Change the trust state of a file"""
+
+    # Save the original permissions of the file
+    orig_perms = os.stat(path).st_mode
+
+    # See if the file is readable
+    try:
+        with open(path):
+            pass
+
+    except IOError:
+        # Try to unlock file to get read access
+        safe_chmod(path, 0o644,
+            'Could not unlock {} for reading'.format(path))
+    if trusted:
+        # Set file to trusted
+        # AKA remove our xattr
+        file_xattrs = xattr.get_all(path)
+        untrusted_attribute = (b'user.qubes.untrusted', b'true')
+
+        # Check if the xattr exists first
+        if untrusted_attribute in file_xattrs:
+            try:
+                xattr.removexattr(path, 'user.qubes.untrusted')
+            except:
+                # Unable to remove our xattr, return original permissions
+                error('Unable to remove untrusted attribute on {}'.format(path))
+                safe_chmod(path, orig_perms,
+                    'Unable to set original perms. on {}'.format(path))
+
+        # Remove visual attributes
+        set_visual_attributes(path, False)
+
+    else:
+        # Set file to untrusted
+        # AKA add our xattr and lock
+        try:
+            xattr.setxattr(path, 'user.qubes.untrusted', 'true')
+            safe_chmod(path, 0o0,
+                    'Unable to set untrusted permissions on: {}'.format(path))
+        except:
+            # Unable to remove our xattr, return original permissions
+            safe_chmod(path, orig_perms,
+                'Unable to return perms after setting as untrusted: {}'.
+                format(path))
+            sys.exit(65)
+
+        # Add visual attributes
+        set_visual_attributes(path, True)
+
+def change_folder(path, trusted):
+    """Change the trust state of a folder"""
+
+    # Remove '/' from end of path
+    path = os.path.normpath(path)
+
+    try:
+        # Create the ~/.config/qubes folder if it doesn't exist
+        home_dir = os.path.expanduser('~')
+        if not os.path.exists(home_dir + '/.config'):
+            os.mkdir(home_dir + '/.config')
+        if not os.path.exists(home_dir + '/.config/qubes'):
+            os.mkdir(home_dir + '/.config/qubes')
```

Comment:
shutil.makedirs

Reply:
`shutil.makedirs()` doesn't [seem to exist](https://docs.python.org/3/library/shutil.html), but `os.makedirs()` does. Changed in 45441f2